File: sessions/2018/easy/1_two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        s = set(nums)
        l = len(nums)
        for i in xrange(l):
            if target - nums[i] in s:
                for j in xrange(i+1, l):
                    if nums[j] == target - nums[i]:
                        return [i, j]

File: pia/2020/easy/1_two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  let indices = [];
  const numsMap = {};

  for (let i = 0; i < nums.length; i++) {
    const currNum = nums[i];
    numsMap[currNum] = i;
  }

  for (let i = 0; i < nums.length; i++) {
    const difference = target - nums[i];
    const doesComplementExist = !!numsMap[difference];
    if (doesComplementExist && numsMap[difference] !== i) {
      return [i, numsMap[difference]];
    }
  }
};
